src/utilities/dates.ts

```diff
@@ -0,0 +1,186 @@
+export enum Weekday {
+  Sunday = 0,
+  Monday = 1,
+  Tuesday = 2,
+  Wednesday = 3,
+  Thursday = 4,
+  Friday = 5,
+  Saturday = 6,
+}
+
+export enum Month {
+  January = 0,
+  February = 1,
+  March = 2,
+  April = 3,
+  May = 4,
+  June = 5,
+  July = 6,
+  August = 7,
+  September = 8,
+  October = 9,
+  November = 10,
+  December = 11,
+}
+
+export interface Range {
+  start: Date;
+  end: Date;
+}
+
+export type Year = number;
+
+export type Week = (Date | null)[];
+
+const WEEK_LENGTH = 7;
+
+export function getWeeksForMonth(
+  month: Month,
+  year: Year,
+  weekStartsOn: Weekday = Weekday.Sunday,
+): Week[] {
+  const firstOfMonth = new Date(year, month, 1);
+  const firstDayOfWeek = firstOfMonth.getDay();
+  const weeks: Week[] = [[]];
+
+  let currentWeek = weeks[0];
+  let currentDate = firstOfMonth;
+
+  const orderedWeekday = getOrderedWeekdays(weekStartsOn);
+  for (let i = 0; i < orderedWeekday.indexOf(firstDayOfWeek); i++) {
+    currentWeek.push(null);
+  }
+
+  while (currentDate.getMonth() === month) {
+    if (currentWeek.length === WEEK_LENGTH) {
+      currentWeek = [];
+      weeks.push(currentWeek);
+    }
+
+    currentWeek.push(currentDate);
+    currentDate = new Date(year, month, currentDate.getDate() + 1);
+  }
+
+  while (currentWeek.length < 7) {
+    currentWeek.push(null);
+  }
+
+  return weeks;
+}
+
+export function dateIsInRange(day: Date | null, range: Range) {
+  if (day == null) {
+    return false;
+  }
+
+  const {start, end} = range;
+
+  return Boolean(start && day > start && end && day < end);
+}
+
+export function dateIsSelected(day: Date | null, range: Range) {
+  if (day == null) {
+    return false;
+  }
+  const {start, end} = range;
+
+  return Boolean(
+    (start && isSameDay(start, day)) || (end && isSameDay(end, day)),
+  );
+}
+
+export function isSameDay(day1: Date, day2: Date) {
```

Comment:
I think we have some already established date helpers in quilt: https://github.com/Shopify/quilt/tree/master/packages/dates#issameday

Reply:
True, however this is a somewhat lazy conversion as I've copied https://github.com/Shopify/javascript-utilities/blob/master/src/dates.ts and removed the functions we weren't using. Happy to move to other libs in follow up PRs, but I didn't want to spend too much time on this right now.